src/Interpreters/getTableExpressions.cpp

```diff
@@ -102,7 +103,11 @@ static NamesAndTypesList getColumnsFromTableExpression(
     {
         auto table_id = context->resolveStorageID(table_expression.database_and_table_name);
         const auto & table = DatabaseCatalog::instance().getTable(table_id, context);
-        auto table_metadata_snapshot = table->getInMemoryMetadataPtr();
+        StorageMetadataPtr table_metadata_snapshot;
+        if (auto * window_view = dynamic_cast<StorageWindowView *>(table.get()))
+            table_metadata_snapshot = window_view->getTargetTable()->getInMemoryMetadataPtr();
```

Comment:
I am in doubt about having this inconsistency of having to make a separate case only for window view in several places. Can this be avoided?

Reply:
A special case for window view in `getTableExpressions` is not a good solution, modifying the code here is for expanding asterisks in `select *` query for windowview. We can replace the columns after `JoinedTables::resolveTables` in `InterpreterSelectQuery::execute`, but I think its also not a good solution